lib/mps_statespace.h

```diff
@@ -155,6 +155,63 @@ class MPSStateSpace {
     }
   }
 
+  // Computes <state1 | state2 > for two equal sized MPS.
+  // Requires: state1.bond_dim() == state2.bond_dim() &&
+  //           state1.num_qubits() == state2.num_qubits()
+  static std::complex<fp_type> InnerProduct(MPS& state1, MPS& state2) {
+    const auto nq = state1.num_qubits();
+    const auto bd = state1.bond_dim();
+    const auto end = Size(state1);
+    auto offset = 0;
+    fp_type* state1_raw = state1.get();
+    fp_type* state2_raw = state2.get();
+
+    // Contract leftmost blocks together, store result in state1 scratch.
+    ConstMatrixMap top((Complex*)state2_raw, 2, bd);
```

Comment:
Add a space between `(Complex*)` and `state` in these two lines?

Reply:
Added.